### PR TITLE
Catch errors at retrieve from NCBI

### PR DESCRIPTION
**Pull request by JuPeg**
I catch exceptions to every error I have found while testing. 
Sometimes no error is thrown but instead of sequences we retrieve error messages. In this case, we also try to retrieve the same batch again. 
I only got the connection reset by peer once, so I just break it, as I couldn't test how to fix it.

### CODE REVIEW

File: tools/fetch_fasta_from_ncbi/retrieve_fasta_from_NCBI.py [hunk 1]
```diff
@@ -166,16 +169,30 @@ def efetch(self, db, query_key, webenv):
...
             except urllib2.HTTPError as e:
+                print "urllib2.HTTPError at ", datetime.now()
                 serverTransaction = False
                 self.logger.info("urlopen error:%s, %s" % (e.code, e.read() ) )
             except httplib.IncompleteRead as e:
+                print "httplib.IncompleteRead at ", datetime.now()
+                serverTransaction = False
+                self.logger.info("IncompleteRead error")
+            except urllib2.URLError as e:
+                print "urllib2.URLError at ", datetime.now()
```
Comment (mvdbeek) [inline tools/fetch_fasta_from_ncbi/retrieve_fasta_from_NCBI.py]: I'm unsure how the logger is configured, but it includes the date by default, so unless the logger is silent, the print statements don't add any information.

File: tools/fetch_fasta_from_ncbi/retrieve_fasta_from_NCBI.py [hunk 2]
```diff
@@ -166,16 +169,30 @@ def efetch(self, db, query_key, webenv):
             try:
                 response = urllib2.urlopen(req)
                 fasta = response.read()
```
Comment (mvdbeek) [inline tools/fetch_fasta_from_ncbi/retrieve_fasta_from_NCBI.py]: I think a better solution is to use
```
if response.getcode() == 200:
    fasta = response.read()
    serverTransaction = True
else:
    continue
```

Normally that should cover all the exceptions you're trying to catch later on,
but I haven't tested this.

**Comment by mvdbeek** [comment]
Looks like a nice addition!

**Comment by mvdbeek** [comment]
It seems that you can provoke connection reset by peer with tcpkill. See https://stackoverflow.com/questions/3139204/triggering-connection-reset-by-peer

**Comment by drosofff** [comment]
@JuPeg if do the proposed changes and commit into this branch, the test should work now

**Comment by JuPeg** [comment]
I didn't have time to test much, but using response.getcode() seems to work fine.

**Pull request closed without merging**